keps/sig-apps/961-maxunavailable-for-statefulset/README.md

```diff
@@ -291,113 +762,284 @@ No, the default behavior remains the same.
-for the controller to converge. This will only happen if after enabling the feature, the customer also sets
-maxUnavailable to a number greater than 1,  but the invariants and the logic will ensure that there are never more than
-maxUnavailable pods with the same identity and never more than maxUnavailable being deleted.
+- This is the ideal scenario where the feature works as intended, allowing users to specify `maxUnavailable` for controlled updates.
+
+4. Feature Disabled on Both kube-apiserver and kube-controller-manager:
+
+- StatefulSet updates revert to the default Kubernetes behavior, ignoring any `maxUnavailable` settings even if previously specified.
+
+In scenarios 1 and 2, running workloads could be impacted due to the mismatch in feature flag settings. For instance, expected rolling update strategies might not be applied, potentially affecting application availability or update speed.
+
+To mitigate these risks, it's crucial to ensure consistent feature flag settings across all control plane components during rollout and to be prepared to adjust configurations based on observed behavior. Monitoring the progress and outcomes of StatefulSet updates closely during the initial adoption phase is recommended to quickly identify and resolve any issues.
 
 ###### What specific metrics should inform a rollback?
-TODO when we reach Beta
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+Administrators should monitor the following metrics to assess the need for a rollback:
+
+- `statefulset_unavailability_violation`: Ths metric reflects the number of times the maxUnavailable condition is violated (i.e. spec.replicas - availableReplicas > maxUnavailable).
+Mutiple violations of maxUnavailable might indicate issues with feature behavior.
 
 ###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
-Will be tested when graduating to Beta.
 
+A manual test was performed, as follows:
+
+1. Create a cluster in 1.23.
+2. Upgrade to 1.24.
+3. Create StatefulSet A with spec.updateStrategy.rollingUpdate.maxUnavailable set to 3, with 6 replicas
+4. Verify a rollout and check if only 3 pods are unavailable at a time ([currently with a bug if podManagementPolicy is set to Parallel](https://github.com/kubernetes/kubernetes/issues/112307))
+5. Downgrade to 1.23.
+6. Verify that the rollout only has 1 pod unavailable at a time, similar to setting maxUnavailable to 1
+7. Create another StatefulSet B not setting maxUnavailable (leaving it nil)
+8. Upgrade to 1.24.
+9. Verify that the rollout has default behavior of only having one pod unavailable at a time
+   Verify that the `maxUnavailable` can be set again to StatefulSet A and test the rollout behavior
+
+TODO:
+A manual test will be performed, as follows:
+
+1. Create a cluster in 1.29.
+2. Upgrade to 1.31.
```

Comment:
In general we'd upgrade to N+1 (instead of N+2). Why are these 2 versions picked?